src/ray/core_worker/common.h

```diff
@@ -0,0 +1,35 @@
+#ifndef RAY_CORE_WORKER_COMMON_H
+#define RAY_CORE_WORKER_COMMON_H
+
+#include <string>
+
+#include "ray/id.h"
+#include "ray/common/buffer.h"
+
+namespace ray {
+
+/// Type of this worker.
+enum class WorkerType { WORKER, DRIVER };
+
+/// Language of Ray tasks and workers.
+enum class Language { PYTHON, JAVA };
+
+/// Information about a remote function.
+struct RayFunction {
+  /// Language of the remote function.
+  const Language language;
+  /// Function descriptor of the remote function.
+  const std::vector<std::string> function_descriptors;
+};
+
+/// Argument of a task.
+struct Arg {
+  /// Id of the argument, if passed by reference, otherwise nullptr.
+  const std::shared_ptr<ObjectID> id;
```

Comment:
I use shared_ptr because either `id` or `data` must be empty. 
Do you prefer using NIL id to indicate emptiness? Then what about data, should we also add a NIL buffer?

Reply:
I see, thanks. It is just not quite intuitive at the same glance for a shared_ptr of ObjectID :)